tenets/codelingo/go/pass-struct-by-pointer/codelingo.yaml

```diff
@@ -0,0 +1,26 @@
+tenets:
+  - name: pass-struct-by-pointer
+    actions:
+      codelingo/review:
+        comment: Passing a pointer to a struct with exported fields such as {{ structName }} is bad practice as it can result in multiple process reading and writing to the same memory.
+    query: |
+      import codelingo/ast/go
+
+      go.type_spec(depth = any):
+        go.ident:
+          name as structName
+        go.struct_type:
+          go.field_list:
+            go.field:
+              go.names:
+                go.ident:
+                  name as fieldName
+                  regex(/^[A-Z]/, fieldName)
```

Comment:
Can we not use the public property here? Assuming this is to only include exported fields